eth/block-validation/api.go

```diff
@@ -228,6 +233,172 @@ func (r *BuilderBlockValidationRequestV2) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+type BlockAssemblerRequest struct {
+	TobTxs             bellatrixUtil.ExecutionPayloadTransactions
+	RobPayload         capellaapi.SubmitBlockRequest
+	RegisteredGasLimit uint64
+}
+
+type IntermediateBlockAssemblerRequest struct {
+	TobTxs             []byte                        `json:"tob_txs"`
+	RobPayload         capellaapi.SubmitBlockRequest `json:"rob_payload"`
+	RegisteredGasLimit uint64                        `json:"registered_gas_limit,string"`
+}
+
+func (b *BlockAssemblerRequest) UnmarshalJSON(data []byte) error {
+	var intermediateJson IntermediateBlockAssemblerRequest
+	err := json.Unmarshal(data, &intermediateJson)
+	if err != nil {
+		return err
+	}
+	err = b.TobTxs.UnmarshalSSZ(intermediateJson.TobTxs)
+	if err != nil {
+		return err
+	}
+	b.RegisteredGasLimit = intermediateJson.RegisteredGasLimit
+	b.RobPayload = intermediateJson.RobPayload
+
+	return nil
+}
+
+// bchain: copied this here to avoid circular dependency
```

Comment:
hmm can you refactor to a shared common package instead ? copying feels a bit bad

Reply:
yeah yeah will do it.